docs/write-a-plugin.md

```diff
@@ -18,9 +18,21 @@ window.$docsify = {
         return content;
       });
 
+      hook.beforeEmbed(function(tokens, next) {
```

Comment:
I don't think we should pass `tokens` produced from a markdown parser to a plugin hook. As @trusktr pointed out, this makes the plugin functionality dependent on a specific markdown library which IMHO is not a good idea.

More importantly, I don't think we should pass "markdown before it becomes HTML" in two different formats (`string` vs `tokens`) to two very similar hooks:

```js
hook.beforeEach(function(content) {
  // ...
  return content;
});

hook.beforeEmbed(function(tokens) {
  // ...
  return tokens;
});
```

If plugins authors want to work with tokens, they can generate their own tokens by accessing Docsify's renderer directly within their plugin:

```js
window.$docsify = {
  plugins: [
    function(hook, vm) {
      hook.beforeEach(function(content) {
        const renderer = vm.config.markdown.renderer

        // Do renderer stuff...

        return content;
      });
    }
  ]
}
```